contracts/IdoContract/src/IdoContract.cs

```diff
@@ -411,79 +422,27 @@ public static bool SwapTokenSecondRound(UInt160 user, UInt160 idoPairContractHas
-            ExecutionEngine.Assert(idoPairContractHash.IsValid && !idoPairContractHash.IsZero, "bad idoPairContractHash");
-            byte[] key = GetUserClaimAmountKey(idoPairContractHash, user);
-            BigInteger oldAmount = GetUserClaimAmountImple(key);
-            amount = amount + oldAmount;
-            ExecutionEngine.Assert(amount >= 0, "bad amount");
-            Storage.Put(Storage.CurrentContext, key, amount);
-        }
-
-        public static BigInteger GetUserClaimAmount(UInt160 idoPairContractHash, UInt160 user)
-        {
-            byte[] key = GetUserClaimAmountKey(idoPairContractHash, user);
-            return GetUserClaimAmountImple(key);
-        }
-
-        private static BigInteger GetUserClaimAmountImple(byte[] key)
-        {
-            ByteString rawAmount = Storage.Get(Storage.CurrentContext, key);
-            return rawAmount is null ? 0 : (BigInteger)rawAmount;
-        }
-
         public static bool ClaimToken(UInt160 user, UInt160 idoPairContractHash)
         {
             ExecutionEngine.Assert(user.IsValid && !user.IsZero, "bad user");
             RegisteredProject project = GetRegisteredProject(idoPairContractHash);
             ExecutionEngine.Assert(project.isNewProject is false, "empty project");
             ExecutionEngine.Assert(project.isReviewed && Ledger.CurrentIndex - project.reviewedHeight - GetVoteTimeSpan() >= GetSwapTimeSpan(), "claim not start");
-            byte[] key = GetUserClaimAmountKey(idoPairContractHash, user);
-            BigInteger oldAmount = GetUserClaimAmountImple(key);
-            ExecutionEngine.Assert(oldAmount > 0, "no unclaimed token");
-            AddUserClaimAmount(idoPairContractHash, user, -oldAmount);
-            SafeTransfer(project.tokenHash, Runtime.ExecutingScriptHash, user, oldAmount);
-            ExecutionEngine.Assert(GetUserClaimAmountImple(key) == 0);
-            return true;
-        }
 
-        public static BigInteger GetRegisteredProjectTotalWeight(UInt160 idoPairContractHash)
-        {
-            byte[] registeredProjectTotalWeightKey = registeredProjectTotalWeightPrefix.Concat(idoPairContractHash);
-            ByteString rawWeight = Storage.Get(Storage.CurrentContext, registeredProjectTotalWeightKey);
-            return rawWeight is null ? 0 : (BigInteger)rawWeight;
-        }
+            ProjectUserInfo projectUserInfo = GetProjectUserInfo(idoPairContractHash, user);
+            ExecutionEngine.Assert(projectUserInfo.claimAmount > 0, "no unclaimed token");
 
-        public static BigInteger GetRegisteredProjectUserWeight(UInt160 idoPairContractHash, UInt160 user)
-        {
-            byte[] userVoteKey = userVotePrefix.Concat(idoPairContractHash).Concat(user);
-            ByteString rawVoted = Storage.Get(Storage.CurrentContext, userVoteKey);
-            return rawVoted is null ? 0 : (BigInteger)rawVoted;
+            SafeTransfer(project.tokenHash, Runtime.ExecutingScriptHash, user, projectUserInfo.claimAmount);
```

Comment:
Do the transfer at the end, after set the flags

Reply:
OK, I will fix it in new pr.